base/standard/src/main/java/org/openscience/cdk/graph/invariant/Canon.java

```diff
@@ -358,7 +362,9 @@ private long primeProduct(int[] ws, long[] ranks, boolean[] hydrogens) {
             label |= Math.abs(chg) & 0x3;
             label <<= 4; // hydrogen count <= 15 (4 bits)
             label |= impH + expH & 0xf;
-
+            label <<= 10;
```

Comment:
No, breaks the SmilesGenerator. Could be an option but we already have the option to use a comparator.